lib/feeder/tasks/sync.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Mix.Tasks.Feeder.Sync do
+  use Mix.Task
+  require Logger
+  import Castle.Redis.Lock
+
+  @shortdoc "Manually run the Feeder data refresh"
+
+  @podlabel "Feeder.SyncPodcasts"
+  @eplabel "Feeder.SyncEpisodes"
+
+  @lock "lock.feeder.sync"
+  @lock_ttl 50
+  @success_ttl 10
+
+  def run(args) do
+    {:ok, _started} = Application.ensure_all_started(:castle)
+
+    lock = Enum.member?(args, "--lock") || Enum.member?(args, "-l")
+    force = Enum.member?(args, "--force") || Enum.member?(args, "-f")
+    all = force || Enum.member?(args, "--all") || Enum.member?(args, "-a")
+
+    pod_since = if force, do: nil, else: Castle.Podcast.max_updated_at()
+    ep_since = if force, do: nil, else: Castle.Episode.max_updated_at()
+
+    if lock do
+      lock "#{@lock}.podcasts", @lock_ttl, @success_ttl, do: sync_podcasts(pod_since, all)
```

Comment:
In addition to locking while working, the `Redis.Lock.lock` function sets a lock ttl _after success_.  This prevents a different server/container with a bit of clock drift from running the same work after it was just done.